display_list/dl_builder.cc

```diff
@@ -1188,6 +1223,22 @@ void DisplayListBuilder::DrawDRRect(const SkRRect& outer,
   drawDRRect(outer, inner);
 }
 void DisplayListBuilder::drawPath(const SkPath& path) {
+  if (!path.isInverseFillType()) {
+    SkRect rect;
+    if (path.isRect(&rect)) {
```

Comment:
This fails because it should ask if the rect is closed and only pass here if it is closed or if we are filling the path.